### PR TITLE
(#4248) fix in CPR with WinSSL

### PR DESCRIPTION
**Pull request by rdeterre**
Specify library name and version:  **cpr/1.5.2**

Here is the pull request that follows issue #4248

I moved the validation code to the `validate()` method. Not sure If I got it right though.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by rdeterre** [comment]
I'm not sure what's the right approach to solve this. Should I just patch CPR v1.3's CMakeLists.txt to add something like `set(ignored ${CMAKE_EXPORT_NO_PACKAGE_REGISTRY})`?

**Comment by prince-chrismc** [comment]
😖 There's no error in the logs

```txt
cpr/1.3.0 (test package): Running test()

----Running------
> bin\test_package
-----------------
status code: 200
headers:     application/json; charset=utf-8
text:        {
  "userId": 1,
  "id": 1,
  "title": "delectus aut autem",
  "completed": false
}
```

**Comment by prince-chrismc** [comment]
close the PR wait 10s and re-open... let's see if it was random 🤞

**Comment by rdeterre** [comment]
It seems to fix v1.3.0 but now v1.4.0 is broken, that seems random indeed ;)

**Comment by prince-chrismc** [comment]
No I think the order might be inconsistent, it tries to do as many versions at once, it's just reporting the first failure

#4261

It might be related to https://github.com/conan-io/conan-center-index/pull/4255/files#diff-aead6257abe8f2d3a654ab6c01f99d3205fea85e598215f04953b3193f6c0b87R106-R107 being moved 🤔

File: recipes/cpr/all/conanfile.py [hunk 1]
```diff
@@ -107,12 +98,19 @@ def _configure_cmake(self):
...
         return self._cmake
 
-    def build(self):
+    def validate(self):
+        if self.options.get_safe("with_winssl", False) and self.settings.os != "Windows":
+            raise ConanInvalidConfiguration("cpr only supports winssl on Windows")
+        if self.options.get_safe("with_openssl", False) and self.options.get_safe("with_winssl", False):
+            raise ConanInvalidConfiguration("cpr can not be built with both openssl and winssl")
+        if self.settings.compiler == "Visual Studio" and self.options.shared and "MT" in self.settings.compiler.runtime:
+            raise ConanInvalidConfiguration("Visual Studio build for shared library with MT runtime is not supported")
```
Comment (prince-chrismc) [inline recipes/cpr/all/conanfile.py:107]:
```suggestion
```

File: recipes/cpr/all/conanfile.py [hunk 2]
```diff
@@ -71,15 +71,6 @@ def configure(self):
             # self.options["libcurl"].with_winssl = True # deprecated in https://github.com/conan-io/conan-center-index/pull/2880
             self.options["libcurl"].with_ssl = "schannel"
```
Comment (prince-chrismc) [inline recipes/cpr/all/conanfile.py:73]:
```suggestion

        if self.options.get_safe("with_winssl", False) and self.settings.os != "Windows":
            raise ConanInvalidConfiguration("cpr only supports winssl on Windows")

        if self.options.get_safe("with_openssl", False) and self.options.get_safe("with_winssl", False):
            raise ConanInvalidConfiguration("cpr can not be built with both openssl and winssl")

        if self.settings.compiler == "Visual Studio" and self.options.shared and "MT" in self.settings.compiler.runtime:
            raise ConanInvalidConfiguration("Visual Studio build for shared library with MT runtime is not supported")

```

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
let's try to revert some changes... I think the new method is a little too new

**Comment by prince-chrismc** [comment]
🤞

**Comment by rdeterre** [comment]
I tried removing `validate()` completely but the build still breaks. I can't see the details of the error, but is there anything I can try to fix this?

**Comment by prince-chrismc** [comment]
CCI has issues at times... when theres no message from the bot. Your best best is to restart CI

You can open close... rename title or edit the top level comment

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**